securesystemslib/storage.py

```diff
@@ -208,14 +217,31 @@ def get(self, filepath:str) -> Iterator[BinaryIO]:
         file_object.close()
 
 
-  def put(self, fileobj: IO, filepath: str) -> None:
+  def put(self, fileobj: IO, filepath: str, restrict: Optional[bool] = False
+        ) -> None:
     # If we are passed an open file, seek to the beginning such that we are
     # copying the entire contents
     if not fileobj.closed:
       fileobj.seek(0)
 
+    # If a file with the same name already exists, the new permissions
+    # may not be applied.
     try:
-      with open(filepath, 'wb') as destination_file:
+      os.remove(filepath)
+    except OSError:
+      pass
+
+    try:
+      if restrict:
+        # On UNIX-based systems restricted files are created with read and
+        # write permissions for the user only (octal value 0o600).
+        fd = os.open(filepath, os.O_WRONLY|os.O_CREAT,
+          stat.S_IRUSR|stat.S_IWUSR)
+      else:
+        # Use the default value (0o777) of the 'mode' argument of os.open()
```

Comment:
Minor comment nits:
Is it worth to mention that passed and default permissions may be silently masked out by umask, or limited to 666 by windows? And is it worth to mention what 666 and 777 means?

Reply:
I'll take a stab at including that info.